src/mount/readdata_cache.h

```diff
@@ -286,8 +286,8 @@ class ReadCache {
 	 * \return cache query result
 	 */
 	void query(Offset offset, Size size, ReadCache::Result &result,
-	           bool insertPending = true) {
-		collectGarbage();
+	           uint32_t usedReadCacheSize, bool insertPending = true) {
```

Comment:
I think your idea was to modify the value of gUsedReadCacheMemory considering the changes of this variable through collectGarbage, maybe you somehow need to pass it as a reference?

Reply:
Done